File: set_1/easy/javascript/020c5507-824a-4bf1-b6b3-c0766fd46c46.js
// Solution 1:

const validateInput = (input) =>{
    const validInputs = ["(", ")", "{", "}", "[", "]"];
  return input.every(elem =>validInputs.includes(elem))
};

const checkParenthesis = val => {
    let valArr = Array.from(val);
    let result, poppedArr = [];
    if(!validateInput(valArr)) {
      return("Input Validation Error!")
    }
    valArr.forEach((elem, index) => {
        
        if (elem === "(") {
            let closingIndex = val.indexOf(")")
            if (closingIndex >= 0) {
                poppedArr.push(valArr[closingIndex]);
                valArr.splice(closingIndex, 1);
            }
            else {
                result = false;
            }
        }
        if (elem === "{") {
            let closingIndex = val.indexOf("}")
            if (closingIndex >= 0) {
                poppedArr.push(valArr[closingIndex]);
                valArr.splice(closingIndex, 1);
            }
            else {
                result = false;
            }
        }
        if (elem === "[") {
            let closingIndex = val.indexOf("]")
            if (closingIndex >= 0) {
                poppedArr.push(valArr[closingIndex]);
                valArr.splice(closingIndex, 1);
            }
            else {
                result = false;
            }
        }
    });
    valArr.length === poppedArr.length ? result = "The parentheses are balanced." : result = "The parentheses are not balanced."
    return result
}

const resp1 = checkParenthesis("{[({})]}");
console.log(resp1);
const resp2 = checkParenthesis("{[({})}");
console.log(resp2);
const resp3 = checkParenthesis("{[({}M)]}");
console.log(resp3);

File: set_1/easy/python/020c5507-824a-4bf1-b6b3-c0766fd46c46.py
# Solution 1:
def validate_input(arr):
    valid_inputs = ["(", ")", "{", "}", "[", "]"]
    return all(elem in valid_inputs  for elem in arr)
def check_parenthesis(val):
    val_arr = []
    val_arr[:0] = val
    result, popped_arr = None, [];
    if(not validate_input(val_arr)):
        return ("Input Validation Error!")
    for i, elem in enumerate(val_arr):
        if(elem == "("):
            closing_index = val_arr.index(")") if "(" in val_arr else -1
            if closing_index >=0:
                popped_arr.append(val_arr[closing_index])
                del val_arr[closing_index]
            else:
                result = False
        if(elem == "{"):
            closing_index = val_arr.index("}") if "{" in val_arr else -1
            if closing_index >=0:
                popped_arr.append(val_arr[closing_index])
                del val_arr[closing_index]
            else:
                result = False
        if(elem == "["):
            closing_index = val_arr.index("]") if "]" in val_arr else -1
            if closing_index >=0:
                popped_arr.append(val_arr[closing_index])
                del val_arr[closing_index]
            else:
                result = False
    result = "The parenthesis are balanced." if len(val_arr) == len(popped_arr) else "The parenthesis are not balanced"
    return result
    

input1, input2, input3 = "{[({})]}", "{[({})}", "{[({}M)]}"
resp1 = check_parenthesis(input1)
print(f"The first input is --> {input1}")
print(resp1)
resp2 = check_parenthesis(input2)
print(f"The second input is --> {input2}")
print(resp2)
resp3 = check_parenthesis(input3)
print(f"The third input is --> {input3}")
print(resp3)

